sys/net/network_layer/sixlowpan/lowpan.c

```diff
@@ -1110,7 +1110,7 @@ uint8_t lowpan_iphc_encoding(int if_id, const uint8_t *dest, int dest_len,
             else if (dest_len == 2 &&
                      ipv6_buf->destaddr.uint32[2] == HTONL(0x000000ff) &&
                      ipv6_buf->destaddr.uint16[6] == HTONS(0xfe00) &&
-                     ipv6_buf->destaddr.uint16[7] == *((uint16_t *) dest)) {
+                     ipv6_buf->destaddr.uint16[7] == (dest[1] << 8 | dest[0])) {
```

Comment:
Using the old byteorder macros causes headaches for me, since I never know when anything is in what order (one reason why I use `network_uintX_t` in the new network stack very stringent and why the code of the old is so confusing and buggy ^^).